programs/castle-vault/src/adapters/port.rs

```diff
@@ -71,34 +103,112 @@ pub struct PortAccounts<'info> {
 impl_has_vault!(PortAccounts<'_>);
 
 impl<'info> LendingMarket for PortAccounts<'info> {
+    fn verify_accounts(&self, program_id: &Pubkey) -> ProgramResult {
```

Comment:
Same comment as in `init_reward_account.rs`: You can do these checks using Anchor constraints in the structs:
#[account(seeds = x, bump = y)]